src/java.base/share/classes/java/lang/foreign/MemoryLayout.java

```diff
@@ -949,7 +954,8 @@ static PathElement sequenceElement() {
          */
         static PathElement dereferenceElement() {
             return new LayoutPath.PathElementImpl(PathKind.DEREF_ELEMENT,
-                    LayoutPath::derefElement);
+                    LayoutPath::derefElement,
+                    "*");
```

Comment:
It seems that this would result in paths like `a.b*` rather than the `*a.b`, which is correct C syntax.

Reply:
Correct. Additional logic is needed to form a correct C syntax. It would be possible to provide a method that does this.

Reply:
We could add such a method under another issue (https://bugs.openjdk.org/browse/JDK-8323746) because it will be much easier to implement it once the contemplated changes for that issue are in.

Reply:
After some more thinking, I think the toString should just mimic the expression used to create the path.
e.g. 
groupElement("foo")
or
sequenceElement()
sequenceElement(2)
sequenceElement(2, 3)
etc.